search/query.go

```diff
@@ -269,22 +269,19 @@ func NewResultsSearchRequest(options SearchRequestOptions) *elastic.SearchReques
 		Explain(options.query.Deb)
 
 	if options.useHighlight {
-
-		//  We use special HighlightQuery with SimpleQueryStringQuery to
-		//	 solve elastic issue with synonyms and highlights.
+		terms := make([]string, 1)
+		if options.query.Term != "" {
+			terms = append(terms, options.query.Term)
+		} else {
+			terms = options.query.ExactTerms
+		}
 
 		contentNumOfFragments := 5 //  elastic default
 		if options.highlightFullContent {
 			contentNumOfFragments = 0
 		}
+		highlightQuery := createHighlightQuery(terms, contentNumOfFragments)
 
-		highlightQuery := elastic.NewHighlight().Fields(
-			elastic.NewHighlighterField("title").NumOfFragments(0).HighlightQuery(elastic.NewSimpleQueryStringQuery(options.query.Term)),
-			elastic.NewHighlighterField("full_title").NumOfFragments(0).HighlightQuery(elastic.NewSimpleQueryStringQuery(options.query.Term)),
-			elastic.NewHighlighterField("description").HighlightQuery(elastic.NewSimpleQueryStringQuery(options.query.Term)),
-			elastic.NewHighlighterField("description.language").HighlightQuery(elastic.NewSimpleQueryStringQuery(options.query.Term)),
-			elastic.NewHighlighterField("content").NumOfFragments(contentNumOfFragments).HighlightQuery(elastic.NewSimpleQueryStringQuery(options.query.Term)),
-			elastic.NewHighlighterField("content.language").NumOfFragments(contentNumOfFragments).HighlightQuery(elastic.NewSimpleQueryStringQuery(options.query.Term)))
 		if !options.partialHighlight {
```

Comment:
put this under createHighlightQuery function

Reply:
[commit](https://github.com/Bnei-Baruch/archive-backend/pull/175/commits/7acf0a50a57a35d025e7f5bb977d161b0ec6aeca)